src/Elastic.Apm.EntityFrameworkCore/EfCoreDiagnosticListener.cs

```diff
@@ -12,9 +11,14 @@ namespace Elastic.Apm.EntityFrameworkCore
 	internal class EfCoreDiagnosticListener : IDiagnosticListener
 	{
 		private readonly IApmAgent _agent;
+		private readonly DbConnectionStringParser _dbConnectionStringParser;
 		private readonly ConcurrentDictionary<Guid, Span> _spans = new ConcurrentDictionary<Guid, Span>();
 
-		public EfCoreDiagnosticListener(IApmAgent agent) => _agent = agent;
+		public EfCoreDiagnosticListener(IApmAgent agent)
+		{
+			_agent = agent;
+			_dbConnectionStringParser = new DbConnectionStringParser(agent.Logger);

```

Comment:
Maybe it's make sense to have `DbConnectionStringParser` at `DbSpanCommon` class?
At first, it will allow not to instantiate parser in each `DbSpanCommon` consumer. Second, it makes sense from responsibility perspective: `DbSpanCommon` is responsible for parsing data from db command and enrich apm span. Why consumers dictate how to parse connection strings?

Reply:
> Maybe it's make sense to have `DbConnectionStringParser` at `DbSpanCommon` class?

Yes, I agree that current implementation is somewhat of hack just to work around `DbSpanCommon` being `static class`. We can make `DbSpanCommon` a regular class, keep an instance of it as a member of some other shared instance (for example `agent.Tracer`) and have an instance of `DbConnectionStringParser` to be a member of `DbSpanCommon`. WDYT?

Reply:
Sound good to me

Reply:
Done.